cli/tests/cfncluster-unittest.py

```diff
@@ -0,0 +1,295 @@
+        self.config_file = "cli/tests/config"
+        self.region = "us-east-1"
+
+
+class CreateClusterArgs(BaseArgs):
+    def __init__(self, template_url, nowait):
+        BaseArgs.__init__(self)
+        self.cluster_name = "test_cluster"
+        self.template_url = template_url
+        self.norollback = True
+        self.nowait = nowait
+
+
+class UpdateClusterArgs(CreateClusterArgs):
+    def __init__(self, template_url, nowait, reset_desired):
+        CreateClusterArgs.__init__(self, template_url=template_url, nowait=nowait)
+        self.reset_desired = reset_desired
+
+
+class CFN_cluster_test(unittest.TestCase):
+    def setUp(self):
+        config_logger_test()
+
+    def test_cfn_cluster_version(self):
+        args = BaseArgs()
+        cfncluster.version(args)
+        log = test_log_stream.getvalue()
+        version_returned = re.match(r"^INFO:\w+\.\w+:(\d+\.\d+\.\d+)$", log).group(1)
+        self.assertEqual(version_returned, version_on_file)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_s3
+    def test_cfn_cluster_create_nowait(self):
+        template_url = setup_configurations()
+        args = CreateClusterArgs(template_url, True)
+        cfncluster.create(args)
+        log = test_log_stream.getvalue()
+        success_message = 'INFO:cfncluster.cfncluster:Status: CREATE_COMPLETE'
+        error_prefix = "CRITICAL:"
+        self.assertTrue(success_message in log)
+        self.assertFalse(error_prefix in log)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_s3
+    def test_cfn_cluster_create_wait(self):
+        template_url = setup_configurations()
+        args = CreateClusterArgs(template_url, False)
+        cfncluster.create(args)
+        log = test_log_stream.getvalue()
+        success_message = 'INFO:cfncluster.cfncluster:MasterPublicIP:'
+        error_prefix = "CRITICAL:"
+        self.assertTrue(success_message in log)
+        self.assertFalse(error_prefix in log)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_s3
+    def test_cfn_cluster_create_fail(self):
+        template_url = setup_configurations()
+        args = CreateClusterArgs('', True)
+        with self.assertRaises(SystemExit) as sys_ex:
+            cfncluster.create(args)
+
+        self.assertEqual(sys_ex.exception.code, 1)
+        log = test_log_stream.getvalue()
+        error_prefix = "CRITICAL:"
+        self.assertTrue(error_prefix in log)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_s3
+    def test_cfn_cluster_list_empty(self):
+        args = BaseArgs()
+        cfncluster.list(args)
+        self.assertEqual(test_log_stream.tell(), 0)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_s3
+    def test_cfn_cluster_list_nonempty(self):
+        template_url = setup_configurations()
+        args = CreateClusterArgs(template_url, False)
+        cfncluster.create(args)
+        cfncluster.list(args)
+        log = test_log_stream.getvalue()
+        self.assertTrue(args.cluster_name in log)
```

Comment:
Can you also test that this is the only thing returned? If we mess up the parsing of cloudformation for example, it could return `cfncluster-cluster_name` instead of just `cluster_name` which will go ahead and throw a "stack not found" exception when the customer goes to list.